pkg/server/build/build.go

```diff
@@ -38,7 +53,64 @@ type CreateOptions struct {
 	Stream    io.Writer
 }
 
-func (ops BuildOperations) CreateByOpts(ctx context.Context, opts *CreateOptions) error {
+func formatPodName(appName, buildName string) string {
+	return fmt.Sprintf("run-%s-build-%s", appName, buildName)
+}
+
+func (ops *BuildOperations) userHasPermission(a *app.App, u *database.User) (bool, error) {
+	teamName, err := ops.appOps.TeamName(a.Name)
+	if err != nil {
+		return false, err
+	}
+	a.Team = teamName
+
+	if !ops.appOps.HasPermission(u, a.Name) {
+		return false, auth.ErrPermissionDenied
+	}
+	return true, nil
+}
+
+func (ops *BuildOperations) Create(ctx context.Context, appName, buildName string, u *database.User, tarBall io.ReadSeeker, runApp bool) (io.ReadCloser, <-chan error) {
+	errChan := make(chan error, 1)
+
+	a, err := ops.appOps.Get(appName)
+	if err != nil {
+		errChan <- err
+		return nil, errChan
+	}
+
+	if uhp, err := ops.userHasPermission(a, u); !uhp || err != nil {
+		errChan <- err
+		return nil, errChan
+	}
```

Comment:
Probably I'm missing something, but isn't `appOps.CheckPermAndGet` enough?

Reply:
You're right! I'll fix it.